clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -19082,6 +19082,15 @@ Value *CodeGenFunction::EmitAMDGPUBuiltinExpr(unsigned BuiltinID,
         CGM.getIntrinsic(Intrinsic::amdgcn_s_sendmsg_rtn, {ResultType});
     return Builder.CreateCall(F, {Arg});
   }
+  case AMDGPU::BI__builtin_amdgcn_make_buffer_rsrc: {
+    llvm::Value *Base = EmitScalarExpr(E->getArg(0));
```

Comment:
I guess you need this just for the pointer type mangling. Can you abuse the CreateUnary* functions for this case?